src/shell/components/FieldTypeDateTime/index.tsx

```diff
@@ -1,33 +1,232 @@
+                      setInputValue(to12HrTime(timeString));
+                    } else {
+                      onChange(`${dateString} ${derivedTime}`);
+                      setIsTimeFieldActive(false);
+                    }
+                  } else if (reason === "selectOption") {
+                    if (typeof time === "object") {
+                      onChange(`${dateString} ${time.value}`);
+                      setIsTimeFieldActive(false);
+                    }
+                  }
+                }}
+                onInputChange={(_, value) => {
+                  setInputValue(value);
+                }}
+                sx={{
+                  width: 96,
+                  "& .MuiAutocomplete-inputRoot": {
+                    py: 0.75,
+                    px: 1,
+
+                    "& input.MuiOutlinedInput-input.MuiAutocomplete-input": {
+                      p: 0,
+                      height: 28,
+                    },
+                  },
+                }}
+                slotProps={{
+                  paper: {
+                    sx: {
+                      width: 184,
+                    },
+                  },
+                }}
+                ListboxProps={{
+                  ref: optionsRef,
+                  sx: {
+                    maxHeight: 180,
+                  },
+                }}
+                renderInput={(params) => (
+                  <TextField
+                    data-cy="dateTimeInputField"
+                    ref={timeFieldRef}
+                    placeholder="HH:MM"
+                    error={invalidInput || error}
+                    onClick={() => {
+                      setIsTimeFieldActive(true);
+                      if (!dateString && !timeString) {
+                        onChange(
+                          `${moment().format("yyyy-MM-DD")} 00:00:00.000000`
+                        );
+                      }
+                    }}
+                    onBlur={() => {
+                      if (!inputValue) {
+                        setInputValue(
+                          to12HrTime(timeString ?? "00:00:00.000000")
+                        );
+                      }
+
+                      setIsTimeFieldActive(false);
+
+                      const derivedTime = toISOString(
+                        getDerivedTime(inputValue)
+                      );
+
+                      if (derivedTime.toLowerCase() === "invalid date") {
+                        // Reset to whatever the last valid time was set to
+                        setInputValue(to12HrTime(timeString));
+                      } else {
+                        onChange(`${dateString} ${derivedTime}`);
+                      }
+                    }}
+                    sx={{
+                      "& .Mui-focused.MuiAutocomplete-inputRoot fieldset.MuiOutlinedInput-notchedOutline":
+                        {
+                          borderColor: invalidInput
```

Comment:
@agalin920 This is so that the border stays red even if the user has the input focused and they're typing in an invalid time format. Default behavior of the input field turns it into orange when focused even if the error prop is true